moped-editor/src/views/projects/projectsListView/helpers.js

```diff
@@ -137,10 +143,35 @@ export const useTableComponents = ({
 /**
  * The Material Table column settings
  */
-export const useColumns = ({ hiddenColumns, classes }) => {
+export const useColumns = () => {
   const location = useLocation();
   const queryString = location.search;
 
+  const [hiddenColumns, setHiddenColumns] = useState(
+    JSON.parse(localStorage.getItem("mopedColumnConfig")) ?? DEFAULT_HIDDEN_COLS
+  );
+
+  const COLUMN_CONFIG = PROJECT_LIST_VIEW_QUERY_CONFIG.columns;
+
+  /*
+   * Sync hidden columns state with local storage
+   */
+  useEffect(() => {
+    localStorage.setItem("mopedColumnConfig", JSON.stringify(hiddenColumns));
+  }, [hiddenColumns]);
+
+  const columnsToReturnInQuery = useMemo(() => {
+    const columnsShownInUI = Object.keys(hiddenColumns)
+      .filter((key) => hiddenColumns[key] === false)
+      .map((key) => key);
+
+    // Some columns are dependencies of other columns to render, so we need to include them
+    // Ex. Rendering ProjectStatusBadge requires current_phase_key which is not a column shown in the UI
+    const columnsNeededToRender = ["current_phase_key"];
```

Comment:
I'm just realizing that this column isn't needed if we aren't showing the **Status** column. I'm torn on whether to try to add extra logic to decide when to add this. Maybe down the road when we have more columns with dependencies?